charts/k8ssandra/templates/reaper/reaper.yaml

```diff
@@ -10,13 +10,11 @@ spec:
     autoScheduling:
       enabled: {{ .Values.reaper.autoschedule }}
     storageType: cassandra
-    jmxUserSecretName: {{ .Release.Name }}-reaper-jmx
+    jmxUserSecretName: {{ include "k8ssandra.reaperJmxUserSecretName" . }}
     cassandraBackend:
-    {{- if .Values.reaper.cassandraUser.secret }}
-      cassandraUserSecretName: {{ .Values.reaper.cassandraUser.secret }}
-    {{- else if or .Values.reaper.cassandraUser.username .Values.cassandra.auth.enabled }}
-      cassandraUserSecretName: {{ include "k8ssandra.clusterName" . }}-reaper
-    {{- end }}
+      {{- if .Values.cassandra.auth.enabled }}
+      cassandraUserSecretName: {{ default (include "k8ssandra.reaperUserSecretName" .) .Values.reaper.cassandraUser.secret }}
```

Comment:
Can you refactor the `reaperUserSecretName` template to be like the `reaperJmxUserSecretName` template? Then this line can be simplified to:

```
cassandraUserSecretName: {{ include "k8ssandra.reaperUserSecretName" . }}
```

We should do the same for the other `k8ssandra.<component>UserSecretName` templates as well. We do this part in a separate, follow up PR though if you prefer.

Reply:
It's not used just here. The behavior was the same in:

https://github.com/k8ssandra/k8ssandra/blob/158a80a154d0e1eee5d0558bc0e1c9fe6166c4e9/charts/k8ssandra/templates/cassandra/cassdc.yaml#L70

And that applies to almost every Cassandra username override

Reply:
I will do a follow up PR for those changes.